### PR TITLE
classes for orders

### PR DESCRIPTION
**Pull request by ppivanov**

### CODE REVIEW

File: POSAPI/src/Models/Orders/Order.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        public string Note { get; set; }
+
+        [Required]
+        public virtual Customer Customer { get; set; }
+
+        public virtual Location DeliveryLocation { get; set; }
+
+        [Required]
+        public OrderType Type { get; set; }
```
Comment (ppivanov) [inline POSAPI/src/Models/Orders/Order.cs:40]: @NNaik98 what do you think about having these as enums instead of objects in the database

Comment (NNaik98) [reply POSAPI/src/Models/Orders/Order.cs:40]: idk what that means lol

Comment (ppivanov) [reply POSAPI/src/Models/Orders/Order.cs:40]: It's just gonna store it as an integer column in the database. User roles right now have a DB table which isn't really necessary.
Every time we try to compare "is user an admin" we need to query the user table AND the role table, but we can get away with using enums I think 🤔

File: POSAPI/src/Controllers/UserController.cs [hunk 2]
```diff
@@ -27,9 +27,8 @@ public ActionResult<string> Register([FromBody] RegisterRequest userDetails)
             if (_model.SystemUsers.CheckInvalidRegisterDetails(userDetails))
                 return BadRequest("Username or Display Name already taken; Please try again");
 
-            var roles = userDetails.GetRoles(_model);
             var userId = _model.GetNewId<SystemUser>();
-            var newUser = new SystemUser(userId, userDetails, roles);
```
Comment (ppivanov) [inline POSAPI/src/Controllers/UserController.cs:32]: getting the user's role directly from userDetails, this was no longer necessary

File: POSAPI/src/Models/Orders/Order.cs (hunk 1 above)
Comment (ppivanov) [reply POSAPI/src/Models/Orders/Order.cs]: removed the role table and replaced it with an enum

**Review by NNaik98 — APPROVED** [review]

**Pull request merged**